File: client/src/remote.ts
export const BASE_URL = "http://localhost:8088"
export const headers = { "Content-Type": "application/json" }

export type Request = "GET" | "POST" | "PUT" | "DELETE"

export interface RequestMethod {
	[key: string]: Request
}

export const requestMethod: RequestMethod = {
	GET: "GET",
	POST: "POST",
	PUT: "PUT",
	DELETE: "DELETE",
}

export interface AppRequest {
	body: string
	headers: {}
	method: string
}

export interface RequestBody {
	id: string
	name: string
	ts: number
}

export const buildRequest = (
	method: string,
	body: Partial<RequestBody>,
): AppRequest => ({
	...baseAppRequest,
	body: JSON.stringify(body),
	headers,
	method,
})

export const baseAppRequest: AppRequest = {
	body: "",
	headers,
	method: requestMethod.GET,
}

// remote shapes are received in snakecase

export interface BeerRemote {
	id: string
	sort_order: number
	name: string
	beer_type: string
	abv: string
	ibu: string
	serving_size: string
	cost: string
	brewery_name: string
	brewery_city: string
	brewery_state: string
	brewery_img_url: string
	keg_id: string
	keg_size: number
	keg_amount_left: number
	updated_ts: number
	created_ts: number
}

File: src/models/remote.rs
use serde_derive::{Deserialize, Serialize};
// all client request models live in this file

// EXAMPLE
// #[derive(Serialize, Deserialize, Debug)]
// pub struct Id {
//     pub id: String,
// }

// #[derive(Serialize, Deserialize, Debug)]
// pub struct PersonReq {
//     pub name: String,
// }

//////////

#[derive(Serialize, Deserialize, Debug)]
pub struct ErrorResponse {
    pub message: String,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct BeerRequest {
    pub name: String,
    pub beer_type: String,
    pub abv: String,
    pub ibu: String,
    pub serving_size: String,
    pub cost: String,
    pub brewery_name: String,
    pub brewery_city: String,
    pub brewery_state: String,
    pub brewery_img_url: String,
    pub keg_id: String,
    pub keg_size: i32,
    pub keg_amount_left: i32,
    pub updated_ts: i32,
    pub created_ts: i32,
}
